aptos-move/framework/supra-framework/sources/supra_governance.move

```diff
@@ -166,6 +171,74 @@ module supra_framework::supra_governance {
         voting_duration_secs: u64,
     }
 
+    /// Configurations of the SupraGovernance, set during Genesis and can be updated by the same process offered
+    /// by this SupraGovernance module.
+    struct SupraGovernanceConfig has key {
+        voting_duration_secs: u64,
+        min_voting_threshold: u64,
+        voters: vector<address>,
+    }
+
+    /// Events generated by interactions with the SupraGovernance module.
+    struct SupraGovernanceEvents has key {
+        create_proposal_events: EventHandle<SupraCreateProposalEvent>,
+        update_config_events: EventHandle<SupraUpdateConfigEvent>,
+        vote_events: EventHandle<SupraVoteEvent>,
+    }
+
+    /// Event emitted when a proposal is created.
+    struct SupraCreateProposalEvent has drop, store {
+        proposer: address,
+        proposal_id: u64,
+        execution_hash: vector<u8>,
+        proposal_metadata: SimpleMap<String, vector<u8>>,
+    }
+
+    /// Event emitted when the governance configs are updated.
+    struct SupraUpdateConfigEvent has drop, store {
+        voting_duration_secs: u64,
+        min_voting_threshold: u64,
+        voters: vector<address>,
+    }
+
+    /// Event emitted when there's a vote on a proposa;
+    struct SupraVoteEvent has drop, store {
+        proposal_id: u64,
+        voter: address,
+        should_pass: bool,
+    }
+
+    #[event]
+    /// Event emitted when a proposal is created.
+    struct SupraCreateProposal has drop, store {
+        proposer: address,
+        proposal_id: u64,
+        execution_hash: vector<u8>,
+        proposal_metadata: SimpleMap<String, vector<u8>>,
+    }
+
+    #[event]
+    /// Event emitted when there's a vote on a proposa;
+    struct SupraVote has drop, store {
+        proposal_id: u64,
+        voter: address,
+        should_pass: bool,
+    }
+
+    #[event]
+    /// Event emitted when the governance configs are updated.
+    struct SupraUpdateConfig has drop, store {
+        voting_duration_secs: u64,
+        min_voting_threshold: u64,
+        voters: vector<address>,
+    }
+
+    /// Used to track which execution script hashes have been approved by governance.
+    /// This is required to bypass cases where the execution scripts exceed the size limit imposed by mempool.
+    struct SupraApprovedExecutionHashes has key {
```

Comment:
Use the existing `ApprovedExecutionHashes` since this is accessed by Rust layer. As it is, this is not exposed via public interface.